### ISSUE TITLE
Reset remote "whoops" command

### ISSUE THREAD

**Issue by ctsstc** [uptech/git-ps-rs#39]
In the scenario that you accidentally sync/push the main branch to the remote with your unvetted patches, it would be nice to have a command that could revert the remote branch to your previous stack. I think this may require pulling and publishing to be done through git ps so that it could track which patches were officially released. Then it could reconcile against its patch stack history to determine that the patches should not be integrated into the remote yet.

This is a dangerous and destructive command (especially if it uses a hard reset), it would be great if it could verify that the remote is not ahead of your local stack, otherwise a larger effort would be required to fix this scenario. I'm not sure if there's any case where your patches could occur at an earlier time than others on the remote -- would the patches have to be plucked out by hand (I assume that's not possible so you would have to reset to before your patches and then cherry pick anything that happened after than that wasn't them?) Hopefully these edge cases aren't as complicated as I'm potentially believing them to be? It would be amazing if it could just trim it out of the history somehow.

**Comment by drewdeponte** [comment]
Yeah, hmm. I am not sure there is a good automated way to deal with this.

It is a relatively easy fix by looking at the tree figuring out the sha of the commit that `origin/main` should be pointing to and then simply running the following.

`git push -f origin <sha-origin-main-should-be-pointing-to>:main`

Obviously the later half is easy to just execute that command. The part that is problematic is figuring which sha `origin/main` should be pointing to.

Also changing history like this can break things for other developers on the team and also should be accompanied by out of band coordination of this effort with the other developers on the team so that they know to handle the divergence appropriately.

**Comment by drewdeponte** [comment]
Not sure I feel like this is really a job for patch stack. Maybe it is just something about Git that people need to be educated about. That is my instinctual feeling. Therefore I am throwing the "education" label on this issue so that we know we need to somehow help educate people about this scenario and what to do if they get into it.

**Comment by ctsstc** [comment]
I agree, this would be a good wiki/markdown article that digs into the options with their pros and cons given the scenario.

**Comment by jazzdragon** [comment]
+1 for this. This happened to me in the past out of absent-mindedness and is easy mistake to do (you are even prompted to do so when doing something like `git status`). The reason why I think it is relevant for this package is that it's forcing you to have local commits that are not on the remote, which can lead to this mistake. Could be a great feature to allow users to have more confidence to use the patch stack knowing that mistakes can be easily ammended.
